File: src/ERP/acme.atena.domain/DTO/Product/CompraProduto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace acme.atena.domain.DTO.Product
{
    public class CompraProduto : AbstractEntityCompetencia
    {
        private decimal _valor;

        public decimal Valor { get => _valor; set => _valor = (value * QuantidadeComprada); }
        public int QuantidadeComprada { get; set; }
        public DateTime DataPagamento { get; set; }
        public bool IsRecebido { get; set; }
        
        public Guid ProdutoId { get; set; }
        public Guid CompraId { get; set; }

        public virtual Produto Produto { get; set; }
        public virtual Compra Compra { get; set; }
        public virtual ICollection<DevolucaoCompra> DevolucaoCompras { get; set; } = new HashSet<DevolucaoCompra>();

    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/product/compra-produto.ts
import { DevolucaoCompra } from './devolucao-compra';
import { Produto } from './produto';
import { AbstractEntityCompetencia } from "../abstract-entity-competencia";
import { Compra } from "./compra";

export class CompraProduto extends AbstractEntityCompetencia{

  public valor: number;
  public quantidadeComprada: number;
  public dataPagamento: Date;
  public isRecebido: boolean;

  public produtoId :string;
  public compraId: string;

  public produto: Produto;
  public compra: Compra;
  public devolucaoCompras: DevolucaoCompra[];


}
